### PR TITLE
deploy: autorelease on tags

### LINKED ISSUE #845: improve release process
What kicks off a release?

`npm version M.m.p && gh pr create` + merge should be enough to deploy all packages & draft release notes.

atm we also need to manually do a release before deployment happens.

### PR DESCRIPTION
**Pull request by casperdcl**
Fixes #845

- [x] auto-draft release notes on tag push
  - facilitates adding release assets easily in future (for #842)
- [x] tidy checkbot
- [x] add tagbot
- [ ] Release method (to be updated in https://cml.dev/doc/contributing/core post-merge):

```diff
  - `git checkout master && git pull && git checkout -b M.m.p && npm version M.m.p && git push && gh pr create`
  - Merge the resulting PR
- - Draft a new [release](https://github.com/iterative/cml/releases), ensuring
-   all commits since last release are summarised in the release notes
+ - `git checkout master && git pull && git tag -f vM.m.p && git push --tags`
+    or comment `/tag vM.m.p SHA` in the PR
+ - Wait for a draft to appear under [releases](https://github.com/iterative/cml/releases)
+ - Check & publish the draft
```

### CODE REVIEW

File: .github/workflows/test-deploy.yml [hunk 1]
```diff
@@ -89,10 +89,24 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - id: meta
+        name: Metadata
+        run: echo ::set-output name=tag::${GITHUB_REF#refs/tags/}
+      - uses: softprops/action-gh-release@v1
+        with:
+          name: CML ${{ steps.meta.outputs.tag }}
+          draft: true
```
Comment (casperdcl) [inline .github/workflows/test-deploy.yml:108]: can easily add `files: build/cml-linux` etc from #842 here

Comment (DavidGOrtega) [reply .github/workflows/test-deploy.yml:108]: how @casperdcl ? with comments?

Comment (casperdcl) [reply .github/workflows/test-deploy.yml:108]: I mean after #847 is merged, we can update #842 to use:

```yml
      - uses: softprops/action-gh-release@v1
        with:
          ...
          files: |
            build/cml-linux
            build/cml-macos
```

Comment (casperdcl) [reply .github/workflows/test-deploy.yml:108]: (updated PR to stop @iterative-olivaw reacting to everything :D)

**Comment by casperdcl** [comment]
we can ignore the failing windows tests since they'll be fixed in #842

**Review by DavidGOrtega — APPROVED** [review]
👍 

Awesome @casperdcl

**Review by 0x2b3bfa0 — APPROVED** [review]
🤷🏼‍♂️ If you prefer this release method, LGTM.

**Comment by casperdcl** [comment]
don't know yet, we'll see how it goes. :lab_coat:

**Pull request merged**